### DISCUSSION TITLE
How to limit the size of a websocket message?

### DISCUSSION THREAD

**Discussion by Zhaoxun** [django/channels#1510] [Q&A] [answered]
Suppose one mallicious client just sent a meaningless websocket message as big as 4GB to the django server, how can the server stops it from sending more websocket frames after realizing it has breached the size limit?

Also, I would like this error to be catched by upper functions and block this client from further logging in after kicking it offline, and record this attack.

Related discussions on stackovervlow:

https://stackoverflow.com/questions/13010354/chunking-websocket-transmission/13049499#13049499
https://stackoverflow.com/questions/47437197/how-to-limit-tornado-websocket-message-size

**Comment by carltongibson** [comment, marked as answer]
My first thought would be to handle this at the Twisted level. Autobahn has [options to limit frame and message sizes](https://autobahn.readthedocs.io/en/latest/websocket/programming.html#common-options-server-and-client). 

For persistence — you'd need to catch errors and then look to keeping a blacklist of IPs (or such). That would need thinking about.

**Reply by Zhaoxun** [reply]
Yes, thank you! As long as it pops up an error and report the connection, it is easy to handle the rest.

**Comment by Mingyang-Han** [comment]
1、As  you say，I already find the maxFramePayloadSize parameter has defined in autobahn\websocket\protocol.py and autobahn\websocket\interfaces.py. Should I modify the parameter in class WebSocketServerFactory(WebSocketFactory) and class WebSocketClientProtocol(WebSocketProtocol) at the same time?
2、If modification is allowed, how should the unit of this parameter be set? Is it byte?
